src/go/configgenerator/filtergen/grpc_field_extraction.go

```diff
@@ -0,0 +1,171 @@
+func NewGRPCFieldExtractionFilterGensFromOPConfig(serviceConfig *confpb.Service, opts options.ConfigGeneratorOptions) ([]FilterGenerator, error) {
+	grpcGen, err := NewGRPCFieldExtractionFilterGenFromOPConfig(serviceConfig, opts, true)
+	if err != nil {
+		return nil, err
+	}
+	if grpcGen == nil {
+		return nil, nil
+	}
+
+	return []FilterGenerator{
+		grpcGen,
+	}, nil
+}
+
+// NewGRPCFieldExtractionFilterGenFromOPConfig creates a single GRPCFielcExtractorGenerator.
+//
+// It also has the option to skip the filter. If enabled, all checks will occur
+// and filter may not be created. Otherwise, filter is always created.
+func NewGRPCFieldExtractionFilterGenFromOPConfig(serviceConfig *confpb.Service, opts options.ConfigGeneratorOptions, maybeSkipFilter bool) (*GRPCFieldExtractionGenerator, error) {
+	var descriptorBin []byte
+	foundDescriptor := false
+
+	for _, sourceFile := range serviceConfig.GetSourceInfo().GetSourceFiles() {
+		configFile := &smpb.ConfigFile{}
+		err := sourceFile.UnmarshalTo(configFile)
+		if err != nil {
+			continue
+		}
+
+		if configFile.GetFileType() == smpb.ConfigFile_FILE_DESCRIPTOR_SET_PROTO {
+			foundDescriptor = true
+			descriptorBin = configFile.GetFileContents()
+		}
+	}
+
+	// Cannot check `descriptorBin == nil` because many tests use empty descriptor.
+	if !foundDescriptor {
+		// b/148605552: Previous versions of the `gcloud_build_image` script did not download the proto descriptor.
+		// We cannot ensure that users have the latest version of the script, so notify them via non-fatal logs.
+		// Log as error instead of warning because error logs will show up even if `--enable_debug` is false.
+		glog.Error("Unable to setup gRPC Field Extraction because no proto descriptor was found in the service config. " +
+			"Please use version 2020-01-29 (or later) of the `gcloud_build_image` script. " +
+			"http://google3/third_party/espv2/source/v12/blob/master/docker/serverless/gcloud_build_image/gcloud_build_image")
+		return nil, nil
+	}
+
+	fds := &descpb.FileDescriptorSet{}
+	if err := proto.Unmarshal(descriptorBin, fds); err != nil {
+		glog.Error("failed to unmarshal protodescriptor, error: ", err)
+		return nil, fmt.Errorf("failed to unmarshal proto descriptor, error: %v", err)
+	}
+	descriptorPolicies := ParseMethodPoliciesFromDescriptorSet(fds, opts.MethodAllowlist)
+	servicePolicies := ParseMethodPoliciesFromServiceConfig(serviceConfig)
+
+	for selector, servicePolicy := range servicePolicies {
+		if descriptorPolicies[selector] == nil {
+			return nil, fmt.Errorf("selector %q from supplementary OP config not found in proto descriptor set, "+
+				"double check method allowlist, api version, or typos", selector)
+		}
+		descriptorPolicies[selector] = servicePolicy
+	}
+
+	for _, methodPolicy := range descriptorPolicies {
+		for _, requestPolicy := range methodPolicy.GetRequestPolicies() {
+			if requestPolicy.GetIsServiceConsumer() || requestPolicy.GetSelector() != "" {
+				break
```

Comment:
This code doesn't do anything.

Also please double check the OSS `policy.proto`. The `is_service_consumer` field is not in 3P yet, it is internal. You cannot use it here.